### DISCUSSION TITLE
error with social cards / Cairo with Python 3.11

### DISCUSSION THREAD

**Discussion by StefanoCecere** [squidfunk/mkdocs-material#4789] [General]
since i have a big site (23 secs to render with Python 3.9), i upgraded to Python 3.11 which with latest MkDocs should be quite faster.

indeed it is!

but i get an error if i enable the social cards, and its relative Cairo lib

before going into details: is anybody using Python 3.11 and social cards? (macOS)

**Comment by squidfunk** [comment]
If it is solely related to Cairo, please raise this to the Cairo maintainers.

**Reply by StefanoCecere** [reply]
sure, already done.. and it's quite difficult to find an answer.. that's why i'm asking first the MkDocs Material community if _anybody_ was using it without problems.

my only solution now is to keep disabled social cards in local macos

**Comment by chrisdburr** [comment]
I'm having a similar issue with using social cards with Python 3.11 and Mac OS 12. No problem when running on a separate Linux machine. 

Have tried using both homebrew and MacPorts to install cairo and the required dependencies, but always get the following error:

```bash
OSError: no library called "cairo-2" was found
no library called "cairo" was found
no library called "libcairo-2" was found
```

Appreciate this is not an issue with MKDocs, but if the community figures out the solution I would also appreciate any guidance as I have also given up on using social cards (or just remove the line from `mkdocs.yml` when testing locally) for the time being due to the amount of time spent troubleshooting.

**Reply by squidfunk** [reply]
Well, the problem is that Cairo is a peer dependency of Pillow, which is one of our upstream dependencies, so this must be raised with the maintainers of Homebrew, Cairo or Pillow. We can't do anything about this library not being available on your platform or system and there's no alternative to Cairo when composing images with Pillow, at least none that I know off. Temporary workaround is to just [disable social cards locally](https://squidfunk.github.io/mkdocs-material/setup/setting-up-social-cards/#+social.cards):

``` yaml
plugins:
  - social:
      cards: !ENV [CI, false]
```

If somebody finds a workaround, please share it – we will then add it to the docs.
